invokeai/app/invocations/latent.py

```diff
@@ -363,9 +360,78 @@ def invoke(self, context: InvocationContext) -> ImageOutput:
                 session_id=context.graph_execution_state_id, node=self
             )
 
+            torch.cuda.empty_cache()
+
             context.services.images.save(image_type, image_name, image, metadata)
             return build_image_output(
-                image_type=image_type,
-                image_name=image_name,
-                image=image
+                image_type=image_type, image_name=image_name, image=image
             )
+
+
+LATENTS_INTERPOLATION_MODE = Literal[
+    "nearest", "linear", "bilinear", "bicubic", "trilinear", "area", "nearest-exact"
+]
+
+
+class ResizeLatentsInvocation(BaseInvocation):
+    """Resizes latents to explicit width/height."""
+
+    type: Literal["lresize"] = "lresize"
+
+    # Inputs
+    latents: Optional[LatentsField]             = Field(description="The latents to resize")
+    width: int                                  = Field(ge=64, multiple_of=8, description="The width to resize to (px)")
+    height: int                                 = Field(ge=64, multiple_of=8, description="The height to resize to (px)")
+    downsample: Optional[int]                   = Field(default=8, ge=1, description="The downsampling factor (leave at 8 for SD)")
```

Comment:
I'm not sure I'd set it up this way. It's a little confusing, even though I understand it based on my experience with the code (width/height here is the desired size, not the latents size). Given how low-level latents are in the code (and as nodes), it might make more sense for these parameters to just represent the actual width/height of the latents tensor (and tell the user these should be 1/8 the size of the final image in the description).

Reply:
Yeah, I was a bit conflicted here. The intention was to avoid requiring the user to doing a bunch of math off-screen but I recognize this limits the usability, adds a bit of magic and introduces a parameter that is unfamiliar to torch users (downsample).

I'm happy to keep it low level for the moment and rely on math nodes or direct user input, though it is less ergonomic.

Reply:
On second thought, if we make w/h the actual dimensions of the tensor, we should do the same for the noise invocation and every other place where we work with latents. This is so unergonomic - we don't want to have to keep multiplying by 8 for everything.

`latents` is a special case of `tensor`, anyways - so I think its best to present the dimensions as pixels and downsample as needed. If we start offering torch functions as nodes, those should definitely be exactly what it says on the tin.

I'm hardcoding the floor division and saying so in the docstring.